keps/79-hierarchical-cohorts/README.md

```diff
@@ -0,0 +1,238 @@
+# KEP-79: Hierarchical Cohorts
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories (Optional)](#user-stories-optional)
+    - [Story 1](#story-1)
+    - [Story 2](#story-2)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+      - [Prerequisite testing updates](#prerequisite-testing-updates)
+    - [Unit Tests](#unit-tests)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+
+Introduce Cohort top-level object to allow setting multi-level quota 
+hierarchy, with advanced borrowing, lending mechanisms.
+
+## Motivation
+
+The current 2-level hierarchy (ClusterQueues and Cohorts) is not expressive
+enough to handle complex use cases of large organizations with tree-like
+team and quota/budget structures.
+
+### Goals
+
+* Create a multi-level hierarchy for advanced quota management. 
+* Be compatible with the existing ClusterQueue API and mechanics.
+* Allow setting constraints about borrowing and lending at all levels.
+* Provide quota for groups of Queues.
+
+### Non-Goals
+
+* Change the existing API and mechanics in a not backward-compatible way.
+* Introduce a new, separate API.
+* Introduce new ways of fair sharing, like ratio-based sharing (at least 
+not in this KEP).
+* Introduce additional preemption models (this will be in a separate KEP).
+
+## Proposal
+
+Introduce a new object called Cohort with the similar quota provisioning mechanism
+as ClusterQueue. Chorots additionally may specify its parent, another Cohort,
+forming together a tree-like organization structure. ClusterQueues will still be able to specify
+the Cohort they belong to. The Cohort mentioned by ClusterQueue doesn't require
+an actual object to be present. If such is not provided, it is understood 
+that the Cohort doesn't provide any quota, has no parent, doesn't belong to any bigger
+structure or has any non-default settings.
+
+The difference between ClusterQuota and Cohort will be that:
+
+* ClusterQueues are leaves in the organization tree, Cohorts are inner nodes. 
+* Cohort doesn't accept any workloads.
+* Quota provided at the Cohort level is to be shared with the entire organization and doesn't 
+  have an owning ClusterQueue.
+* Borrowing limit specified at the Cohort level means that the entire subtree cannot borrow more 
+  from the rest of the organization tree than the given value.
+* Lending limit specified at the Cohort levels means that the rest of the organization tree
+  cannot borrow more from the subtree than the given value.
+
+### User Stories (Optional)
```

Comment:
worth putting some yaml snippets to gather feedback.

Reply:
Added descriptions how the proposal handles user stories. Yamls would be unnecessarily long.